### PR TITLE
[DEV-879] Preserve spaces during token processing

### PR DESCRIPTION
**Pull request by marcobottaro**
#### List of Changes

Preserve spaces during token processing

#### Motivation and Context

When converting tables from GitBook as cards, the parser removed all spaces, but sometimes they are needed to break up words inside other tags (strong, link, etc.)

#### Screenshots (if appropriate):
before: 
image

after:
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/gitbook-docs/src/markdoc/tokenProcessor.ts [hunk 1]
```diff
@@ -29,12 +29,19 @@ const makeHtmlParser = () => {
       result.push(token);
     },
     ontext: (content) => {
-      if (typeof content === 'string' && content.trim().length > 0) {
+      if (typeof content !== 'string') return;
+      if (content.trim().length > 0) {
```
Comment (datalek) [inline packages/gitbook-docs/src/markdoc/tokenProcessor.ts:33]: What about to replace the `else if` with the following condition?
```suggestion
      if (content.trim().length > 0 || content === ' ') {
```

File: packages/gitbook-docs/src/__tests__/parseContent.test.ts [hunk 2]
```diff
@@ -110,6 +110,94 @@ describe('parseContent', () => {
...
+                ]),
+              ]
+            ),
+            new Markdoc.Tag('StyledText', { style: 'strong' }, ['?']),
+          ]),
+          new Markdoc.Tag('CardItem', {}, []),
+          new Markdoc.Tag('CardItem', {}, []),
+        ]),
+      ]),
+    ]);
```
Comment (datalek) [inline packages/gitbook-docs/src/__tests__/parseContent.test.ts:200]: What about to simplify as following? It should cover all the cases we need

```suggestion
  });

  it('should preserve a single space into a td tag nested into other tags', () => {
    expect(
      parseContent(
        '<td><strong>before space</strong> <a><strong>after space</strong></a></td>',
        config
      )
    ).toStrictEqual([
      new Markdoc.Tag('StyledText', { style: 'strong' }, ['before space']),
      ' ',
      new Markdoc.Tag('Link', {}, [
        new Markdoc.Tag('StyledText', { style: 'strong' }, ['after space']),
      ]),
    ]);
```

**Review by datalek — COMMENTED** [review]
Good job! I've just suggested some simplification

**Review by kin0992 — APPROVED** [review]
👍

**Pull request merged**